File: src/main/java/com/micabytes/ink/StoryInterrupt.java
package com.micabytes.ink;

import org.jetbrains.annotations.NonNls;

public interface StoryInterrupt {

  @NonNls String getId();

  @NonNls String getInterrupt();

  @NonNls String getInterruptCondition();

  @NonNls String getInterruptFile();

  boolean isChoice();

  boolean isDivert();

  boolean isActive();

  void done();

}

File: src/main/kotlin/com/micabytes/ink/StoryInterrupt.kt
package com.micabytes.ink

abstract class StoryInterrupt(
    val id: String,
    val text: String,
    val condition: String,
    val file: String) {
  val isChoice = text.startsWith(Symbol.CHOICE_DOT) || text.startsWith(Symbol.CHOICE_PLUS)
  val isDivert = text.startsWith(Symbol.DIVERT)
  var isActive: Boolean = true
}
